packages/frontend/src/pages/campaigns/index.tsx

```diff
@@ -56,14 +56,14 @@ export const Campaigns = () => {
     const [ordering, setOrdering] = useState<SelectOption>(ORDERING_OPTIONS[0]);
     const [state, setState] = useState<SelectOption>(STATE_OPTIONS[0]);
 
-    // filter and sort results
+    // filter results
     const filteredTokens = useMemo(() => {
         return filterKPITokens(Object.values(kpiTokens));
     }, [kpiTokens]);
-
+    // sort results
     const sortedFilteredTokens = useMemo(() => {
-        return sortKPITokens(filteredTokens);
-    }, [filteredTokens]);
+        return sortKPITokens(filteredTokens, ordering.label);
```

Comment:
Maybe it's clearer to pass the `value` to the `sortKPITokens`; this way we avoid the `ordering?.toLowerCase() === "oldest"`.

One more thing, I'd use the values `-1` as `descending` and `1` as `ascending` for the sorting.

Reply:
Agree on both points. We can also work with strings directly if we make the values `newest` and `oldest` respectively instead of `1` and `2`, and then the `sortKPITokens` can take them in. A bit like it was before. I think it's way more elegant.